packages/swap/test/ord-sale.test.ts

```diff
@@ -0,0 +1,218 @@
+  if (i === 1) return 'mock:1111111111111111111111111111111111111111111111111111111111111111'
+  if (i === 2) return 'mock:2222222222222222222222222222222222222222222222222222222222222222'
+  if (i === 3) return 'mock:3333333333333333333333333333333333333333333333333333333333333333'
+  if (i === 4) return 'mock:4444444444444444444444444444444444444444444444444444444444444444'
+  if (i === 5) return 'mock:5555555555555555555555555555555555555555555555555555555555555555'
+  throw new Error('getTestTxId is only defined for parameters smaller than 6.')
+}
+
+export function getTestRev(txId = 0, outNum = 0): string {
+  return `${getTestTxId(txId)}:${outNum}`
+}
+
+class PaymentMock {
+  _id: string
+  _rev: string
+  _root: string
+  _amount: number
+  _owners: string[]
+
+  constructor(owner: string, amount: number) {
+    const r = Math.floor(Math.random() * 1000)
+    this._id = getTestRev(0, r)
+    this._rev = getTestRev(0, r)
+    this._root = getTestRev(0, r)
+    this._owners = [owner]
+    this._amount = amount
+  }
+
+  transfer(to: string) {
+    this._owners = [to]
+  }
+}
+
+class ValuableMock {
+  _id: string
+  _rev: string
+  _root: string
+  _amount: number
+  _owners: string[]
+
+  constructor() {
+    const r = Math.floor(Math.random() * 100000)
+    this._id = getTestRev(1, r)
+    this._rev = getTestRev(1, r)
+    this._root = getTestRev(1, r)
+    this._owners = [randomPublicKey]
+    this._amount = 7860
+  }
+
+  setAmount(amount: number) {
+    this._amount = amount
+  }
+}
+
+describe('Ord Sale', () => {
+  let tx: any
+  let sellerPublicKey: string
+  const nftPrice = 1e8
+  const fee = 1e6
+
+  describe('Creating an NFT and an offer to sell', () => {
+    let nft: NFT
+    const seller = new Computer(RLTC)
+    sellerPublicKey = seller.getPublicKey()
+
+    before("Fund Seller's wallet", async () => {
+      await seller.faucet(1e8)
+    })
+
+    it('Seller creates an NFT', async () => {
+      nft = await seller.new(NFT, ['name', 'symbol'])
+      // @ts-ignore
+      expect(nft).to.matchPattern({ name: 'name', symbol: 'symbol', ...meta })
+    })
+
+    it('Seller creates a swap transaction for the NFT with the desired price', async () => {
+      const b1Mock = new ValuableMock()
+      const b2Mock = new ValuableMock()
+      const paymentMock = new PaymentMock(seller.getPublicKey(), nftPrice)
+      const { SIGHASH_SINGLE, SIGHASH_ANYONECANPAY } = Transaction
+
+      ;({ tx } = await seller.encode({
+        exp: `${Sale} Sale.exec(b1, b2, nft, payment)`,
+        env: { b1: b1Mock._rev, b2: b2Mock._rev, nft: nft._rev, payment: paymentMock._rev },
+        mocks: { b1: b1Mock, b2: b2Mock, payment: paymentMock },
+        // eslint-disable-next-line no-bitwise
+        sighashType: SIGHASH_SINGLE | SIGHASH_ANYONECANPAY,
+        inputIndex: 2,
+        fund: false,
+      }))
+    })
+
+    it('The third input has been signed by seller, the second input is unsigned', () => {
```

Comment:
It must say here: "'The third input has been signed by seller, the forth input (payment) is unsigned"